### PR TITLE
gromacs: add adh_dodec, change nsteps to variable

### PR DESCRIPTION
**Pull request by danielahlin**
Also remove resethway in favour of resetstep

### CODE REVIEW

**Comment by danielahlin** [comment]
Make the gromacs command, subcommand and most currently used parameters variables:

gmx: Name of the gromacs binary - to be able to support both gmx_mpi (for gromacs+mpi) and gmx (for gromacs~mpi)

grompp: prefix for how to run grompp. This is primarily to be able to use mpirun with -n 1 for mpi builds where applications needs to be run with mpirun even for -n 1

mdrun: Primarily to complement the grompp variable

verbose: To turn on and of verbose mode (it usually has a slight performance impact)

nopmetune: To control the parameter with the same name

dlb: Value for dynamic load balancing (yes or no)

**Comment by rfbgo** [comment]
Hey Daniel! 

PR is looking good, main blocker now is the style. Our report hides the useful bit in the middle, so here it is incase it's useful

```
Step #1 - "ramble-style-tests": var/ramble/repos/builtin/applications/gromacs/application.py:123: [E124] closing bracket does not match visual indentation
Step #1 - "ramble-style-tests": var/ramble/repos/builtin/applications/gromacs/application.py:127: [E124] closing bracket does not match visual indentation
Step #1 - "ramble-style-tests": var/ramble/repos/builtin/applications/gromacs/application.py:131: [E124] closing bracket does not match visual indentation
Step #1 - "ramble-style-tests": var/ramble/repos/builtin/applications/gromacs/application.py:135: [E124] closing bracket does not match visual indentation
Step #1 - "ramble-style-tests": var/ramble/repos/builtin/applications/gromacs/application.py:139: [E124] closing bracket does not match visual indentation
Step #1 - "ramble-style-tests": var/ramble/repos/builtin/applications/gromacs/application.py:143: [E124] closing bracket does not match visual indentation
Step #1 - "ramble-style-tests": var/ramble/repos/builtin/applications/gromacs/application.py:147: [E124] closing bracket does not match visual indentation
Step #1 - "ramble-style-tests": var/ramble/repos/builtin/applications/gromacs/application.py:151: [E124] closing bracket does not match visual indentation
Step #1 - "ramble-style-tests": 
```

We value the contribution and are happy to help out with fixing the style stuff, just let us know

**Comment by danielahlin** [comment]
Thanks for the feedback, hopefully addressed now.

File: var/ramble/repos/builtin/applications/gromacs/application.py [hunk 1]
```diff
@@ -98,6 +98,49 @@ class Gromacs(SpackApplication):
...
+    workload_variable('mdrun', default='{gmx} mdrun',
+                      description='How to run mdrun',
+                      workloads=all_workloads)
+    workload_variable('nsteps', default=str(20000),
+                      description='Simulation steps',
+                      workloads=all_workloads)
+    workload_variable('resetstep', default='{str(int(0.9*{nsteps}))}',
+                      description='Reset performance counters at this step',
+                      workloads=all_workloads)
+    workload_variable('verbose', default="-v",
```
Comment (douglasjacobsen) [inline var/ramble/repos/builtin/applications/gromacs/application.py:135]: Given verbose mode has a slight performance impact, can we default it to off?

You can even add:
`values=['', '-v']` to tell people how to turn it on.

**Review by douglasjacobsen — CHANGES_REQUESTED** [review]
One small request, otherwise this LGTM.

File: var/ramble/repos/builtin/applications/gromacs/application.py (hunk 1 above)
Comment (danielahlin) [reply var/ramble/repos/builtin/applications/gromacs/application.py]: Ok - sounds reasonable. I added valid values to some other variables as well.

For verbose there may be a dependency between the value and what FOMs that are available (if some output  from verbose mode is used for FOMs).

**Review by douglasjacobsen — APPROVED** [review]

**Pull request merged**